examples/benchmark.py

```diff
@@ -31,183 +124,469 @@ def downloadAmberSuite():
             else:
                 integ = mm.LangevinMiddleIntegrator(temperature, friction, dt)
-        positions = pdb.positions
-        system = ff.createSystem(pdb.topology, nonbondedMethod=method, nonbondedCutoff=cutoff, constraints=constraints, hydrogenMass=hydrogenMass)
-    if options.ensemble == 'NPT':
-        system.addForce(mm.MonteCarloBarostat(1*unit.bar, temperature, 100))
-    print('Step Size: %g fs' % dt.value_in_unit(unit.femtoseconds))
+        elif options.bond_constraints == 'allbonds':
+            dt = 0.005*unit.picoseconds
+            hydrogenMass = 4*unit.amu
+            if options.ensemble == 'NVE':
+                integ = mm.VerletIntegrator(dt)
+            else:
+                integ = mm.LangevinIntegrator(temperature, friction, dt)
+        else:
+            raise ValueError(f'Unknown bond_constraints {options.bond_constraints}')
+
+    test_result['timestep_in_fs'] = dt.value_in_unit(unit.femtoseconds)
     properties = {}
     initialSteps = 5
-    if options.device is not None and platform.getName() in ('CUDA', 'OpenCL'):
+    if options.device is not None and options.platform in ('CUDA', 'OpenCL'):
         properties['DeviceIndex'] = options.device
         if ',' in options.device or ' ' in options.device:
             initialSteps = 250
-    if options.precision is not None and platform.getName() in ('CUDA', 'OpenCL'):
+    if options.precision is not None and options.platform in ('CUDA', 'OpenCL'):
         properties['Precision'] = options.precision
 
-    # Run the simulation.
-    
+    # Add barostat if requested
+    if options.ensemble == 'NPT':
+        system.addForce(mm.MonteCarloBarostat(1*unit.bar, temperature, 100))
+
+    # Create the Context
+    platform = mm.Platform.getPlatformByName(options.platform)
     integ.setConstraintTolerance(1e-5)
     if len(properties) > 0:
         context = mm.Context(system, integ, platform, properties)
     else:
         context = mm.Context(system, integ, platform)
+
+    # Store information about the Platform used by the Context
+    platform = context.getPlatform()
+    test_result['precision'] = options.precision # requested precision
+    test_result['platform'] = platform.getName()
+    test_result['platform_properties'] = { property_name : platform.getPropertyValue(context, property_name) for property_name in platform.getPropertyNames() }
+
+    # Prepare the simulation
     context.setPositions(positions)
     if amber:
-        if inpcrd.boxVectors is not None:
-            context.setPeriodicBoxVectors(*inpcrd.boxVectors)
         mm.LocalEnergyMinimizer.minimize(context, 100*unit.kilojoules_per_mole/unit.nanometer)
     context.setVelocitiesToTemperature(temperature)
 
+    # Take one step (to enforce constraints and trigger kernel compulation) and store the initial state
+    integ.step(1)
+    state = context.getState(getPositions=True, getVelocities=True, getEnergy=True, getForces=True, getParameters=True)
```

Comment:
This is unnecessary.  `timeIntegration()` already takes care of it.

Reply:
It's necessary to ensure subsequent serialization for Folding@home benchmark WU generation will match the same initial conditions for local benchmarking.

Reply:
`timeIntegration()` already takes some initial steps to ensure everything is warmed up.  Since those steps are nondeterministic (assuming constant temperature), the starting state at the end of them will be different either way.  But that also makes no difference for benchmarking.  The speed difference for very slightly different starting states is at the noise level.